ibis/expr/operations.py

```diff
@@ -299,20 +420,8 @@ class Least(CoalesceLike):
 
 
 class Abs(UnaryOp):
-
-    """
-    Absolute value
-    """
-
-    output_type = rules.type_of_arg(0)
-
-
-def _ceil_floor_output(self):
-    arg = self.args[0]
-    if isinstance(arg, ir.DecimalValue):
-        return arg._factory
-    else:
-        return rules.shape_like(arg, 'int64')
+    """Absolute value"""
+    output_type = rlz.typeof('arg')
```

Comment:
Can you not pass `arg` directly here? It *is* in scope after all.

Reply:
`arg` is actually an `Argument` instance without any knowledge of the underlying data, also doesn't know its name until the metaclass finishes signature construction.
A solution would be to set the argument's name in the metaclass or use descriptors and `_prefixed` slots.

Reply:
Actually You could give me a couple of ideas how should we abstract `return_` definition.